### PR TITLE
Remove deprecated `PipelineRun.Spec.ServiceAccountNames` field.

### PR DESCRIPTION
**Pull request by abayer**
# Changes

Finishing up #2614

We had multiple ways to specify the `serviceAccountName` for `PipelineTask`s - the original `PipelineRun.Spec.ServiceAccountNames`, and the more general `PipelineRun.Spec.TaskRunSpecs`, which also allows specifying a pod template, metadata, and container overrides for individual steps and sidecars. Therefore, we deprecated `ServiceAccountNames`, and are now removing it.

This has been scheduled for removal since May 2021.

/kind cleanup

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in
(if there are no user facing changes, use release note "NONE")

# Release Notes

```release-note
Removes deprecated `PipelineRun.Spec.ServiceAccountNames` field; use `PipelineRun.Spec.TaskRunSpecs` instead.
```

### CODE REVIEW

File: docs/pipelineruns.md [hunk 1]
```diff
@@ -645,7 +643,7 @@ Consult the documentation of the custom task that you are using to determine whe
 
 ### Mapping `ServiceAccount` credentials to `Tasks`
 
-If you require more granularity in specifying execution credentials, use the `serviceAccountNames` field to
+If you require more granularity in specifying execution credentials, use the `taskRunSpecs[].taskServiceAccountName` field to
```
Comment (lbernick) [inline docs/pipelineruns.md:646]: not relevant to this PR but maybe in v1 we can change this to `taskRunSpecs[].serviceAccountName`

File: pkg/apis/pipeline/v1beta1/pipelinerun_types.go [hunk 2]
```diff
@@ -174,14 +174,8 @@ func (pr *PipelineRun) HasTimedOut(ctx context.Context, c clock.PassiveClock) bo
...
 // PipelineTask if configured, otherwise it returns the PipelineRun's serviceAccountName.
-func (pr *PipelineRun) GetServiceAccountName(pipelineTaskName string) string {
-	serviceAccountName := pr.Spec.ServiceAccountName
-	for _, sa := range pr.Spec.ServiceAccountNames {
-		if sa.TaskName == pipelineTaskName {
-			serviceAccountName = sa.ServiceAccountName
-		}
-	}
-	return serviceAccountName
+func (pr *PipelineRun) GetServiceAccountName() string {
```
Comment (lbernick) [inline pkg/apis/pipeline/v1beta1/pipelinerun_types.go:177]: same as above, this can be removed

**Review by lbernick — APPROVED** [review]

File: pkg/apis/pipeline/v1beta1/pipelinerun_types.go (hunk 2 above)
Comment (abayer) [reply pkg/apis/pipeline/v1beta1/pipelinerun_types.go:177]: Ah, my old "don't change anything you don't have to" instinct going overboard... =) On it.

**Review by jerop — APPROVED** [review]
similar to the other removal of deprecated fields, this may also need an email to users and contributors - thank you @abayer!

/lgtm

**Pull request merged**